### PR TITLE
pallet: Implement rollup-queue

### PR DESCRIPTION
**Pull request by kvinwang**
This PR implements the off-chain rollup queue according to the given API design in https://github.com/Phala-Network/phala-blockchain/pull/1073.

### CODE REVIEW

**Review by h4x3rotab — APPROVED** [review]
LGTM

File: pallets/offchain-rollup/src/types.rs [hunk 1]
```diff
@@ -25,4 +25,5 @@ pub enum Cond {
 #[derive(Debug, PartialEq, Eq, Encode, Decode, Clone, scale_info::TypeInfo)]
 pub enum Action {
 	Reply(ActionBytes),
+	QueueHeadSet(u32),
```
Comment (h4x3rotab) [inline pallets/offchain-rollup/src/types.rs:28]:
```suggestion
	SetQueueHead(u32),
```

Make it a verb?

**Review by h4x3rotab — APPROVED** [review]
LGTM

File: pallets/offchain-rollup/src/types.rs (hunk 1 above)
Comment (kvinwang) [reply pallets/offchain-rollup/src/types.rs]: Done

**Pull request merged**